Index: src/Session3/FileManager.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+>package Session3;\n\npublic class FileManager {\n}\n
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/Session3/FileManager.java	(revision 33c8d773782927d053ddee9f5c3cc030768abe70)
+++ src/Session3/FileManager.java	(date 1612617419563)
@@ -1,4 +1,45 @@
 package Session3;
 
+import java.io.*;
+import java.util.Scanner;
+
 public class FileManager {
+
+    public String writeObject(Object obj) {
+        Scanner sc = new Scanner(System.in);
+        System.out.println("Enter filepath:");
+        String filepath = sc.nextLine();
+
+        try {
+            FileOutputStream outFile = new FileOutputStream(filepath);
+            ObjectOutputStream outObject = new ObjectOutputStream(outFile);
+            outObject.writeObject(obj);
+            outObject.close();
+            System.out.println("Student: " + obj + " created succesfully, and saved to: " + filepath);
+
+            return filepath;
+        }
+        catch (IOException e) {
+            e.printStackTrace();
+            return null;
+        }
+    }
+
+    public Object readObject(String filepath) {
+        try {
+            FileInputStream inFile = new FileInputStream(filepath);
+            ObjectInputStream inObject = new ObjectInputStream(inFile);
+
+            Object obj = inObject.readObject();
+
+            System.out.println("The Object has been read from the file");
+            inObject.close();
+            return obj;
+
+        } catch (Exception ex) {
+            ex.printStackTrace();
+            return null;
+        }
+    }
+
 }
Index: src/Session3/Student.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/Session3/Student.java	(date 1612617710639)
+++ src/Session3/Student.java	(date 1612617710639)
@@ -0,0 +1,68 @@
+package Session3;
+
+import java.io.*;
+import java.util.Arrays;
+
+public class Student implements Serializable {
+
+        //default serialVersion id
+        private static final long serialVersionUID = 1L;
+
+        private String first_name;
+        private String last_name;
+        private int age;
+
+        public Student(String fname, String lname, int age){
+            this.first_name = fname;
+            this.last_name  = lname;
+            this.age = age;
+        }
+
+        public void setFirstName(String fname) {
+            this.first_name = fname;
+        }
+
+        public String getFirstName() {
+            return this.first_name;
+        }
+
+        public void setLastName(String lname) {
+            this.first_name = lname;
+        }
+
+        public String getLastName() {
+            return this.last_name;
+        }
+
+        public void setAge(int age) {
+            this.age = age;
+        }
+
+        public int getAge() {
+            return this.age;
+        }
+
+        @Override
+        public String toString() {
+            return " First Name: " + this.first_name +
+                    " Last Name : " + this.last_name + " Age : " + this.age;
+        }
+
+
+    public static void main(String[] args) {
+        FileManager fManager = new FileManager();
+
+        Student[] students = {new Student("Queen", "Elizabeth", 42),
+                              new Student("David", "Attenborough", 42),
+                              new Student("Jimmy", "Carr", 69)};
+
+        String filepath = fManager.writeObject(students);
+        Object obj = fManager.readObject(filepath);
+
+        System.out.println("From file " + filepath + ":");
+
+        for (Student student : (Student[]) obj) {
+            System.out.println(student);
+        }
+    }
+}
